src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -66,6 +66,7 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             services.AddStandbyServices();
 
             // Core script host services
+            services.TryAddSingleton<IScriptJobHostEnvironment, WebScriptJobHostEnvironment>();
```

Comment:
Where this service was being registered before, it was null at resolution time when the host was running. The test I added also demonstrated that - service was null before my fix and the test failed.

Reply:
I think the intention of this was that it doesn't live at this layer and that if you need it, you pull it from the current host. Looking at the implementation, I don't see why this wouldn't work, though. @fabiocav may have better insights into that.

Reply:
That's accurate. This was meant to be `JobHost` scoped (and each JobHost would have its own scoped instance). Changing the registration here changes the semantic and makes this behave like `IWebScriptJobHostEnvironment`, which is the environment service that represents the web host/application. 

This service is also what things like the CLI would inject if they need to override restart/shutdown behavior.